pkg/cloudprovider/providers/aws/device_allocator.go

```diff
@@ -40,11 +45,27 @@ type DeviceAllocator interface {
 	// name. Only the device suffix is returned, e.g. "ba" for "/dev/xvdba".
 	// It's up to the called to add appropriate "/dev/sd" or "/dev/xvd" prefix.
 	GetNext(existingDevices ExistingDevices) (mountDevice, error)
+
+	// LockForUse will lock the device allocator, so as another goroutine can't use it
+	LockForUse()
+
+	// Mark for use makes last selected device as taken
+	MarkForUse()
```

Comment:
We can't use same lock. If we use `attachingMutex` lock to safeguard `deviceAllocator` index across `performAwsAttach` function - we will end up locking entire cluster whilst API call that does `Attach` returns. 

After this change - we are using per node locks. Even without this change, device allocation was always protected using `c.attachingMutex` but the lock was released before making API call.

After this change, lock in `deviceAllocator` is only released after API call. Also the lock is released before `waitForAttachmentStatus`. So we don't really hold the lock while we wait for volume to attach. We just hold the lock - for first API call.